lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -41,385 +44,379 @@ export type LoadTokenOptions = {
-            request.correlationId || BrowserCrypto.createNewGuid();
-
-        const idTokenClaims = response.id_token
-            ? AuthToken.extractTokenClaims(response.id_token, base64Decode)
-            : undefined;
-
-        const authorityOptions: AuthorityOptions = {
-            protocolMode: this.config.system.protocolMode,
-            knownAuthorities: this.config.auth.knownAuthorities,
-            cloudDiscoveryMetadata: this.config.auth.cloudDiscoveryMetadata,
-            authorityMetadata: this.config.auth.authorityMetadata,
-        };
-        const authority = request.authority
-            ? new Authority(
-                  Authority.generateAuthority(
-                      request.authority,
-                      request.azureCloudOptions
-                  ),
-                  this.config.system.networkClient,
-                  this.storage,
-                  authorityOptions,
-                  this.logger,
-                  request.correlationId || BrowserCrypto.createNewGuid()
-              )
-            : undefined;
-
-        const cacheRecordAccount: AccountEntity = await this.loadAccount(
-            request,
-            options.clientInfo || response.client_info || "",
-            correlationId,
-            idTokenClaims,
-            authority
+export async function loadExternalTokens(
+    config: BrowserConfiguration,
+    request: SilentRequest,
+    response: ExternalTokenResponse,
+    options: LoadTokenOptions
+): Promise<AuthenticationResult> {
+    const operatingContext = new StandardOperatingContext(config);
+    if (!operatingContext.isBrowserEnvironment()) {
+        throw createBrowserAuthError(
+            BrowserAuthErrorCodes.nonBrowserEnvironment
         );
+    }
 
-        const idToken = await this.loadIdToken(
-            response,
-            cacheRecordAccount.homeAccountId,
-            cacheRecordAccount.environment,
-            cacheRecordAccount.realm,
-            correlationId
-        );
+    const correlationId =
+        request.correlationId || BrowserCrypto.createNewGuid();
+
+    const idTokenClaims = response.id_token
+        ? AuthToken.extractTokenClaims(response.id_token, base64Decode)
+        : undefined;
+
+    const authorityOptions: AuthorityOptions = {
+        protocolMode: config.system.protocolMode,
+        knownAuthorities: config.auth.knownAuthorities,
+        cloudDiscoveryMetadata: config.auth.cloudDiscoveryMetadata,
+        authorityMetadata: config.auth.authorityMetadata,
+    };
+    const cryptoOps = new CryptoOps(
+        operatingContext.logger,
+        config.telemetry.client
+    );
+    const storage = new BrowserCacheManager(
+        config.auth.clientId,
+        config.cache,
+        cryptoOps,
+        operatingContext.logger,
+        config.telemetry.client,
+        new EventHandler(operatingContext.logger),
+        buildStaticAuthorityOptions(config.auth)
+    );
+    const logger = operatingContext.logger;
```

Comment:
Let's skip the operating context here
```suggestion
    const logger = new Logger(config.system.loggerOptions || {});
```

Reply:
Done!